server/consumer.go

```diff
@@ -4150,7 +4150,8 @@ func (o *consumer) decStreamPending(sseq uint64, subj string) {
 	// If it was pending process it like an ack.
 	// TODO(dlc) - we could do a term here instead with a reason to generate the advisory.
 	if wasPending {
-		o.processAckMsg(sseq, p.Sequence, rdc, false)
+		// We could have lock for sytream so do this in a go routine.
```

Comment:
little typo: `systream`. But more importantly: is it going to be an issue if those "acks" are processed possibly out of order? Imagine a fast pace ingestion with low stream limits (even possibly KV use-case) that cause the "first" message to be removed, so those go routines may first: start to accumulate, second: processed "out of order".

Since you are now changing to a processTerm, then maybe the comment above "wasPending" can be revisited: also, it means that now an advisory is going to be generated for every message that is dropped. Is that ok?

Also, when invoked from storeUpdates(), we have something like:
```
...
    for _, o := range cl {
        o.decStreamPending(seq, subj)
    }
}
if mset.jsa != nil {
	mset.jsa.updateUsage(mset.tier, mset.stype, bd)
}
```
Previously, the ack processing was done in place, now it is not and so is that not possible that updateUsage() will have wrong information?

Reply:
I think term is more appropriate here, that is what we are really doing.

The jsa update could be wrong, but quickly will be corrected.. IMO

Reply:
Ok then